app/models/CreateGoodsRecordRequest.scala

```diff
@@ -20,27 +20,52 @@ import cats.data.{EitherNec, NonEmptyChain}
 import cats.implicits._
 import pages._
 import play.api.libs.json.{Json, OFormat}
+import queries.CommodityQuery
 
-final case class GoodsRecord(
+import java.time.Instant
+
+final case class CreateGoodsRecordRequest(
+  eori: String,
   actorId: String,
-  traderReference: String,
-  commodityCode: String,
+  traderRef: String,
+  comcode: String,
+  goodsDescription: String,
   countryOfOrigin: String,
-  goodsDescription: String
+  category: Int,
+  assessments: Option[String],
+  supplementaryUnit: Option[Int],
+  measurementUnit: Option[String],
+  comcodeEffectiveFromDate: Instant,
+  comcodeEffectiveToDate: Option[Instant]
 )
 
-object GoodsRecord {
+object CreateGoodsRecordRequest {
 
-  implicit lazy val format: OFormat[GoodsRecord] = Json.format
+  implicit lazy val format: OFormat[CreateGoodsRecordRequest] = Json.format
 
-  def build(answers: UserAnswers, eori: String): EitherNec[ValidationError, GoodsRecord] =
+  def build(answers: UserAnswers, eori: String): EitherNec[ValidationError, CreateGoodsRecordRequest] =
     (
       Right(eori),
       answers.getPageValue(TraderReferencePage),
-      getCommodityCode(answers),
       answers.getPageValue(CountryOfOriginPage),
-      getGoodsDescription(answers)
-    ).parMapN(GoodsRecord.apply)
+      getGoodsDescription(answers),
+      getCommodity(answers)
+    ).parMapN((eori, traderReference, countryOfOrigin, goodsDescription, commodity) =>
+      CreateGoodsRecordRequest(
```

Comment:
we want two separate objects, one for GoodsRecord and another for the request  - the request

Reply:
i was thinking about this - i guess the proper good record object would be created with the get goods record ticket - but i don't need it for mine so i didn't include it

Reply:
Exactly, I will pick up from here on the response and see if it fits in nicely - for now, great

Reply:
added in the router stuff, but still have my own separate create